sphinx_gallery/binder.py

```diff
@@ -45,20 +45,39 @@ def gen_binder_url(fname, binder_conf):
     """
     # Build the URL
     fpath_prefix = binder_conf.get('filepath_prefix')
-    binder_fpath = '_downloads/{}'.format(replace_py_ipynb(fname))
+    link_base = binder_conf.get('notebooks_folder', '_downloads')
+    if link_base != '_downloads':
+        # In this case, we want to keep the relative path to sub-folders
+        # Split so we have the relative path starting after the gallery dir
+        relative_link = _get_gallery_dir_path(fname, gallery_conf)
+        path_link = os.path.join(
+            link_base, replace_py_ipynb(relative_link))
+    else:
+        # Assume that we're in the root of _downloads
+        fname = os.path.basename(fname)
```

Comment:
why do you need this extra `basename`?

Reply:
I believe that fname here is a full path, so this just grabs the name of the file itself